contracts/src/libraries/PolynomialEval.sol

```diff
@@ -150,26 +165,7 @@ library PolynomialEval {
             return 0;
         }
 
-        uint256[8] memory localDomainElements;
-        uint256 groupGen = self.groupGen;
-        assembly {
-            let tmp := 1
-            mstore(localDomainElements, tmp) // g^0
-            tmp := mulmod(tmp, groupGen, p)
-            mstore(add(localDomainElements, 0x20), tmp) // g^1
-            tmp := mulmod(tmp, groupGen, p)
-            mstore(add(localDomainElements, 0x40), tmp) // g^2
-            tmp := mulmod(tmp, groupGen, p)
-            mstore(add(localDomainElements, 0x60), tmp) // g^3
-            tmp := mulmod(tmp, groupGen, p)
-            mstore(add(localDomainElements, 0x80), tmp) // g^4
-            tmp := mulmod(tmp, groupGen, p)
-            mstore(add(localDomainElements, 0xa0), tmp) // g^5
-            tmp := mulmod(tmp, groupGen, p)
-            mstore(add(localDomainElements, 0xc0), tmp) // g^6
-            tmp := mulmod(tmp, groupGen, p)
-            mstore(add(localDomainElements, 0xe0), tmp) // g^7
-        }
+        uint256[8] memory localDomainElements = self.domainElements;
```

Comment:
this is extra memory copying, we can directly access them in the logic below via `self.domainElements` instead of via `localDomainElements`

Reply:
done in [5708d75](https://github.com/EspressoSystems/espresso-sequencer/pull/1860/commits/5708d754124d98b3f1472ae5919633cb8d7419a0)

Reply:
Thanks!